lib/api/apiUtils/quotas/quotaUtils.js

```diff
@@ -0,0 +1,248 @@
+                        }
+                        return parallelDone();
+                    });
+                }
+                return parallelDone();
+            },
+            accountQuota: parallelDone => {
+                if (accountQuota > 0 && account?.account) {
+                    return QuotaService.getUtilizationMetrics('account',
+                        account.account, null, {
+                        action: apiName,
+                        ..._inflights,
+                    }, (err, accountMetrics) => {
+                        if (err) {
+                            return parallelDone(err);
+                        }
+                        if (!isMetricStale(accountMetrics, 'account', account.account, apiName, inflight, log) &&
+                            accountMetrics.bytesTotal + _inflightForCheck > accountQuota) {
+                            log.debug('Account quota exceeded', {
+                                accountId: account.account,
+                                action: apiName,
+                                inflight,
+                                quota: accountQuota,
+                                bytesTotal: accountMetrics.bytesTotal,
+                            });
+                            accountQuotaExceeded = true;
+                        }
+                        return parallelDone();
+                    });
+                }
+                return parallelDone();
+            },
+        }, err => {
+            if (err) {
+                return done(err);
+            }
+            const removeInflights = _done => async.parallel({
+                deleteBucketInflights: parallelDone => {
+                    if (bucketQuota > 0) {
+                        return QuotaService.getUtilizationMetrics('bucket',
+                            `${bucket.getName()}_${creationDate}`, null, {
+                            action: apiName,
+                            ..._inflights,
+                        }, parallelDone);
+                    }
+                    return parallelDone();
+                },
+                deleteAccountInflights: parallelDone => {
+                    if (accountQuota > 0 && account?.account) {
+                        return QuotaService.getUtilizationMetrics('account',
+                            account.account, null, {
+                            action: apiName,
+                            ..._inflights,
+                        }, parallelDone);
+                    }
+                    return parallelDone();
+                },
+            }, _done);
+            if (_inflights) {
+                _inflights.inflight = -_inflights.inflight;
+            }
+            if (shouldSendInflights) {
+                // eslint-disable-next-line no-param-reassign
+                request.removeQuotaEvaluationInflights = {
+                    ...request.removeQuotaEvaluationInflights,
+                    [apiName]: removeInflights,
```

Comment:
* it seems to be it may be simpler and cleaner to just introduce a callback called in any case at the end of the request : and let this callback decide if it needs to clear the inflights or not.
* what is the point of indexing by `apiName` ? we should have only one API name for each request...

Reply:
This was changed to a table as discussed: it also has the benefit of handling in a clean way multiple actions having the same name (e.g., during multi object deletes).